dspace/config/spring/api/core-services.xml

```diff
@@ -143,70 +148,206 @@
         </property>
     </bean>
 
+    <!-- Config like this will tell our VirtualMetadataPopulator to include the virtual metadata field
+         'dc.contributor.author' on the appropriate item with the values defined in the value-ref.
+          This value-ref should be a bean of type VirtualBean -->
     <util:map id="isAuthorOfPublicationMap">
-        <entry key="dc_contributor_author" value-ref="publicationAuthor_author"/>
+        <entry key="dc.contributor.author" value-ref="publicationAuthor_author"/>
     </util:map>
-    <util:list id="publicationAuthor_author">
-        <value>orgunit.identifier.name</value>
-        <value>person.identifier.lastname</value>
-        <value>person.identifier.firstname</value>
-    </util:list>
+    <!-- This Concatenate bean will take all the values of each metadata field configured in the list
+         and it will join all of these together with the separator defined in this bean. This means that whichever
+         entry this bean belongs to, that metadata field will have the value of the related item's metadata values
+         joined together with this separator.
+         If the related item has:
+            person.identifier.lastname = Smith
+            person.identifier.firstname = John
+         Then the original item will have, in this case:
+            dc.contributor.author = Smith, John -->
+    <bean class="org.dspace.content.virtual.Concatenate" id="publicationAuthor_author">
+        <property name="fields">
+            <util:list>
+                <value>person.identifier.lastname</value>
+                <value>person.identifier.firstname</value>
+            </util:list>
+        </property>
+        <property name="separator">
+            <value>, </value>
+        </property>
+    </bean>
 
+    <!-- Config like this will tell our VirtualMetadataPopulator to include the virtual metadata field
+     'dc.contributor.other' on the appropriate item with the values defined in the value-ref.
+      This value-ref should be a bean of type VirtualBean -->
     <util:map id="isOrgUnitOfPublicationMap">
         <entry key="dc_contributor_other" value-ref="publicationOrgUnit_name"/>
     </util:map>
-    <util:list id="publicationOrgUnit_name">
-        <value>orgunit.identifier.name</value>
-    </util:list>
+    <bean class="org.dspace.content.virtual.Collected" id="publicationOrgUnit_name">
+        <property name="fields">
+            <util:list>
+                <value>orgunit.identifier.name</value>
+            </util:list>
+        </property>
+    </bean>
 
     <util:map id="isPersonOfProjectMap">
         <entry key="project_contributor_author" value-ref="projectPerson_author"/>
     </util:map>
-    <util:list id="projectPerson_author">
-        <value>person.identifier.lastname</value>
-        <value>person.identifier.firstname</value>
-    </util:list>
+    <!-- This Concatenate bean will take all the values of each metadata field configured in the list
```

Comment:
I realized I was the one to ask for more inline comments in this config file. However, I don't think we need to repeat the same exact `Concatenate` comments/example multiple times in the same file.  I'd rather we put this example into JavaDocs (as `Concatenate` has no specific Javadocs) where we can link to it from this config file.

Reply:
I've moved the general comments to the JavaDoc of the class, and only left the specific comments explaining the use case for that configuration